src/jit/emitarm64.cpp

```diff
@@ -7991,19 +8042,19 @@ BYTE*               emitter::emitOutputLJ(insGroup  *ig, BYTE *dst, instrDesc *i
     }
     else if (loadLabel)
     {
-        if (fmt == IF_LS_1A)       // LS_1A   XX......iiiiiiii iiiiiiiiiiittttt      Rt simm21
+        if (fmt == IF_LS_1A)       // LS_1A   XX...V..iiiiiiii iiiiiiiiiiittttt      Rt simm21
         {
             // INS_ldr or INS_ldrsw (PC-Relative)
 
             // Is the target a vector register?
             if (isVectorRegister(id->idReg1()))
             { 
-                code &= 0x3FFFFFFF;                                  // clear the size bits
-                code |= insEncodeDatasizeVLS(code, id->idOpSize());  // XX
+                code |= insEncodeDatasizeVLS(code, id->idOpSize());  // XX V
                 code |= insEncodeReg_Vt(id->idReg1());               // ttttt
             }
             else
             {
+                code |= insEncodeDatasizeLS(code, id->idOpSize());   // XX
```

Comment:
If we are going to use insEncodeDatasizeLS to set the datasize bits here then we will need to update this method.
The current implementation start off by checking bit 23 which is part of the literal forms immediate.

I agree that using this method to set the  datasize bits is what we want to do.

The method needs to be expanded to check if bit 29 is zero and then encode the 64-bit size.
But you also have to check if bit 31 is set (as it will be for ldrsw pc-rel or literal) before setting bit 30.


Note that the encoding has to handle three instructioLDRSW,